src/System.Data.SqlClient/tests/ManualTests/SQL/UdtTest/SqlServerTypesTest.cs

```diff
@@ -0,0 +1,308 @@
+                }
+            }
+        }
+
+        [CheckConnStrSetupFact]
+        public static void TestUdtSqlDataReaderGetStreamSequentialAccess()
+        {
+            TestUdtSqlDataReaderGetStream(CommandBehavior.SequentialAccess);
+        }
+
+        [CheckConnStrSetupFact]
+        public static void TestUdtSqlDataReaderGetStream()
+        {
+            TestUdtSqlDataReaderGetStream(CommandBehavior.Default);
+        }
+
+        private static void TestUdtSqlDataReaderGetStream(CommandBehavior behavior)
+        {
+            using (SqlConnection connection = new SqlConnection(DataTestUtility.TcpConnStr))
+            {
+                connection.Open();
+                SqlCommand command = connection.CreateCommand();
+                command.CommandText = "select hierarchyid::Parse('/1/1/3/') as col0, geometry::Parse('LINESTRING (100 100, 20 180, 180 180)') as col1, geography::Parse('LINESTRING(-122.360 47.656, -122.343 47.656)') as col2";
+                using (SqlDataReader reader = command.ExecuteReader(behavior))
+                {
+                    Assert.True(reader.Read());
+
+                    MemoryStream buffer = null;
+                    byte[] bytes = null;
+
+                    buffer = new MemoryStream();
+                    using (Stream stream = reader.GetStream(0))
+                    {
+                        stream.CopyTo(buffer);
+                    }
+                    bytes = buffer.ToArray();
+                    Assert.Equal("5ade", ToHexString(bytes));
+
+                    buffer = new MemoryStream();
+                    using (Stream stream = reader.GetStream(1))
+                    {
+                        stream.CopyTo(buffer);
+                    }
+                    bytes = buffer.ToArray();
+                    Assert.Equal("0000000001040300000000000000000059400000000000005940000000000000344000000000008066400000000000806640000000000080664001000000010000000001000000ffffffff0000000002", ToHexString(bytes));
+
+                    buffer = new MemoryStream();
+                    using (Stream stream = reader.GetStream(2))
+                    {
+                        stream.CopyTo(buffer);
+                    }
+                    bytes = buffer.ToArray();
+                    Assert.Equal("e610000001148716d9cef7d34740d7a3703d0a975ec08716d9cef7d34740cba145b6f3955ec0", ToHexString(bytes));
+
+                    if (behavior == CommandBehavior.Default)
+                    {
+                        buffer = new MemoryStream();
+                        using (Stream stream = reader.GetStream(0))
+                        {
+                            stream.CopyTo(buffer);
+                        }
+                        bytes = buffer.ToArray();
+                        Assert.Equal("5ade", ToHexString(bytes));
+                    }
+                }
+            }
+        }
+
+        [CheckConnStrSetupFact]
+        public static void TestUdtSchemaMetadata()
```

Comment:
Can we have a test for Udt as a parameter ? This can be handled in a separate PR though.

Reply:
Ignore this. Github review hadn't expanded the Diffs for a few tests. I see that there are tests for Udts in SqlParameter